contracts/integrations/test/fuzz_tests/staking_rewards_test.ts

```diff
@@ -0,0 +1,125 @@
+import { blockchainTests } from '@0x/contracts-test-utils';
+
+import { Actor } from '../framework/actors/base';
+import {
+    MakerTaker,
+    OperatorStakerMaker,
+    StakerKeeper,
+    StakerMaker,
+    StakerOperator,
+} from '../framework/actors/hybrids';
+import { Keeper } from '../framework/actors/keeper';
+import { Maker } from '../framework/actors/maker';
+import { PoolOperator } from '../framework/actors/pool_operator';
+import { Staker } from '../framework/actors/staker';
+import { Taker } from '../framework/actors/taker';
+import { filterActorsByRole } from '../framework/actors/utils';
+import { AssertionResult } from '../framework/assertions/function_assertion';
+import { BlockchainBalanceStore } from '../framework/balances/blockchain_balance_store';
+import { DeploymentManager } from '../framework/deployment_manager';
+import { Simulation, SimulationEnvironment } from '../framework/simulation';
+import { Pseudorandom } from '../framework/utils/pseudorandom';
+
+import { PoolMembershipSimulation } from './pool_membership_test';
+import { StakeManagementSimulation } from './stake_management_test';
+
+export class StakingRewardsSimulation extends Simulation {
+    protected async *_assertionGenerator(): AsyncIterableIterator<AssertionResult | void> {
+        const { actors } = this.environment;
+        const stakers = filterActorsByRole(actors, Staker);
+        const keepers = filterActorsByRole(actors, Keeper);
+
+        const poolMembership = new PoolMembershipSimulation(this.environment);
+        const stakeManagement = new StakeManagementSimulation(this.environment);
+
+        const actions = [
+            ...stakers.map(staker => staker.simulationActions.validWithdrawDelegatorRewards),
+            ...keepers.map(keeper => keeper.simulationActions.validFinalizePool),
+            ...keepers.map(keeper => keeper.simulationActions.validEndEpoch),
+            poolMembership.generator,
+            stakeManagement.generator,
+        ];
+        while (true) {
+            const action = Pseudorandom.sample(actions);
+            yield (await action!.next()).value; // tslint:disable-line:no-non-null-assertion
+        }
+    }
+}
+
+blockchainTests('Staking rewards fuzz test', env => {
```

Comment:
Do we need a `.skip` here? (I recall it was previously included w other fuzz tests).

Reply:
nah the `before` block ensures that it only runs if the `FUZZ_TEST` env variable is set